panel/models/comm_manager.ts

```diff
@@ -106,7 +112,7 @@ export class CommManager extends Model {
     }
```

Comment:
> regarding `t.document is null`

`document` may be `null` here. This would be evident if `this.ns.shared_views` was typed. What changed in 3.0, making this code fail, is `attach_document()` is called during model initialization (to fully initialize models in a single pass) and not after, which in turn may make `_document_listener` trigger during object graph reconstruction from JSON. At least this is my rough analysis of this failure. At least `view.document?. ...` is in order, but the logic may still be faulty.